src/Analysis/Ast/Impl/Extensions/PythonModuleExtensions.cs

```diff
@@ -26,5 +27,39 @@ internal static PythonAst GetAst(this IPythonModule module)
             => ((IAstNodeContainer)module).GetAstNode<T>(o);
         internal static void AddAstNode(this IPythonModule module, object o, Node n)
             => ((IAstNodeContainer)module).AddAstNode(o, n);
+
+        /// <summary>
+        /// Returns the string line corresponding to the given line number
+        /// </summary>
+        /// <param name="lineNum">The line number</param>
+        internal static string GetLine(this IPythonModule module, int lineNum) {
+            string content = module.Analysis?.Document?.Content;
+            if (string.IsNullOrEmpty(content)) {
+                return string.Empty;
+            }
+
+            SourceLocation source = new SourceLocation(lineNum, 1);
+            var start = module.GetAst().LocationToIndex(source);
+            var end = start;
+
+
+            for (; end < content.Length && content[end] != '\n' && content[end] != '\r'; end++) ;
+            return content.Substring(start, end - start).Trim('\t', ' ');
+        }
+
+        /// <summary>
+        /// Returns the comment corresponding to the given line or null if there is no comment
+        /// </summary>
+        /// <param name="lineNum">The line number</param>
+        internal static string GetComment(this IPythonModule module, int lineNum) {
+            string line = module.GetLine(lineNum);
+
+            int commentPos = line.IndexOf('#');
+            if (commentPos < 0) {
+                return null;
```

Comment:
return `string.Empty` for safety

Reply:
Was initially worried about distinguishing between the case where no comment exists and a comment exists but it has nothing after #. Do you think this is worth distinguishing?

Reply:
Probably not, honestly. If we need to check that something is missing a comment altogether, then we can modify the function then, but it's much more likely that we're going to be checking for what's in a comment rather than if one doesn't exist.

Reply:
Sure then, I'll change this.